packages/react/src/StateLabel/StateLabel.tsx

```diff
@@ -33,97 +29,120 @@ const octiconMap = {
+  }
+
+  &[data-variant='normal'] {
+    padding-inline: 12px;
+    padding-block: var(--base-size-8, ${get('space.2')});
+    font-size: var(--text-body-size-medium, ${get('fontSizes.1')});
+  }
+
+  /* Status */
+  &[data-status='issueClosed'] {
+    background-color: var(--bgColor-done-emphasis, ${get('done.emphasis')});
+  }
+
+  &[data-status='issueClosedNotPlanned'] {
+    background-color: ${get('colors.neutral.emphasis')};
+  }
+
+  &[data-status='pullClosed'] {
+    background-color: ${get('colors.closed.emphasis')};
+  }
+
+  &[data-status='pullMerged'] {
+    background-color: ${get('colors.done.emphasis')};
+  }
+
+  &[data-status='pullQueued'] {
+    background-color: ${get('colors.attention.emphasis')};
+  }
+
+  &[data-status='issueOpened'] {
+    background-color: ${get('colors.open.emphasis')};
+  }
+
+  &[data-status='pullOpened'] {
+    background-color: ${get('colors.open.emphasis')};
+  }
+
+  &[data-status='draft'] {
+    background-color: ${get('colors.neutral.emphasis')};
+  }
+
+  &[data-status='issueDraft'] {
+    background-color: ${get('colors.neutral.emphasis')};
+  }
+
+  &[data-status='unavailable'] {
+    background-color: ${get('colors.neutral.emphasis')};
+  }
+
+  /* Icon */
+  & .StateLabelIcon {
+    margin-inline-end: var(--base-size-4, ${get('space.1')});
+  }
+
+  &[data-variant='small'] .StateLabelIcon {
+    width: 1em;
+  }
+
   ${sx};
 `
 
-export type StateLabelProps = ComponentProps<typeof StateLabelBase>
+export type StateLabelProps = React.ComponentPropsWithoutRef<typeof StateLabelBase> & {
+  variant?: 'small' | 'normal'
+  status: keyof typeof octiconMap
+} & SxProp
 
-function StateLabel({children, status, variant: variantProp = 'normal', ...rest}: StateLabelProps) {
-  const octiconProps = variantProp === 'small' ? {width: '1em'} : {}
+function StateLabel({children, status, variant = 'normal', ...rest}: StateLabelProps) {
+  const Icon = octiconMap[status]
+  const label = getIconLabel(status)
   return (
-    <StateLabelBase {...rest} variant={variantProp} status={status}>
-      {/* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */}
-      {status && <Octicon {...octiconProps} icon={octiconMap[status] || QuestionIcon} sx={{mr: 1}} />}
+    <StateLabelBase {...rest} data-status={status} data-variant={variant}>
+      <Icon aria-label={label ?? undefined} className="StateLabelIcon" />
       {children}
     </StateLabelBase>
   )
 }
 
+function getIconLabel(status: keyof typeof octiconMap): string | null {
+  switch (status) {
+    case 'issueOpened':
+      return 'Issue Opened'
+    case 'issueClosed':
+      return 'Issue Closed'
+    case 'issueDraft':
+      return 'Draft issue'
+    case 'issueClosedNotPlanned':
+      return 'Issue Closed, Not Planned'
+    case 'pullOpened':
+      return 'Pull Request Opened'
+    case 'pullClosed':
+      return 'Pull Request Closed'
+    case 'pullMerged':
+      return 'Pull Request Merged'
```

Comment:
I don't think we need everything caps!

`Pull request merged` or `Pull Request merged` depending on if we cap request?